main.go

```diff
@@ -190,6 +203,163 @@ func main() {
+
+	if err := addNewlyFormattedInitialisedVolumeNames(ctx, client, setupLog); err != nil {
+		setupLog.Error(err, "problem patching Initialised volumes")
+		os.Exit(1)
+	}
+}
+
+func addNewlyFormattedInitialisedVolumeNames(ctx context.Context, client crClient.Client, setupLog logr.Logger) error {
+	aeroClusterList := &asdbv1.AerospikeClusterList{}
+	if err := client.List(ctx, aeroClusterList); err != nil {
+		if errors.IsNotFound(err) {
+			// Request objects not found.
+			return nil
+		}
+		// Error reading the object.
+		return err
+	}
+
+	for idx := range aeroClusterList.Items {
+		aeroCluster := aeroClusterList.Items[idx]
+
+		podList, err := getClusterPodList(ctx, client, &aeroCluster)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Request objects not found.
+				return nil
+			}
+			// Error reading the object.
+			return err
+		}
+
+		var patches []jsonpatch.PatchOperation
+
+		for podIdx := range podList.Items {
+			pod := podList.Items[podIdx]
+			initializedVolumes := aeroCluster.Status.Pods[pod.Name].InitializedVolumes
+			initializedVolumesMap := make(map[string]string)
+			oldFormatInitVolNames := make([]string, 0, len(initializedVolumes))
+
+			for idx := range initializedVolumes {
+				initVolInfo := strings.Split(initializedVolumes[idx], "@")
+				if len(initVolInfo) < 2 {
+					oldFormatInitVolNames = append(oldFormatInitVolNames, initializedVolumes[idx])
+				} else {
+					initializedVolumesMap[initVolInfo[0]] = initVolInfo[1]
+				}
+			}
+
+			for idx := range oldFormatInitVolNames {
+				if _, ok := initializedVolumesMap[oldFormatInitVolNames[idx]]; !ok {
+					pvcUID, pvcErr := getPVCUid(context.TODO(), client, &pod, initializedVolumes[idx])
+					if pvcErr != nil {
+						return pvcErr
+					}
+
+					// Appending volume name as <vol_name>@<pvcUID> in initializedVolumes list
+					initializedVolumes = append(initializedVolumes, fmt.Sprintf("%s@%s", initializedVolumes[idx], pvcUID))
+				}
+			}
+
+			if len(initializedVolumes) > len(aeroCluster.Status.Pods[pod.Name].InitializedVolumes) {
+				setupLog.Info("Got updates initialised volumes list", "initvol", initializedVolumes, "pod-name", pod.Name)
+				patch1 := jsonpatch.PatchOperation{
+					Operation: "replace",
+					Path:      "/status/pods/" + pod.Name + "/initializedVolumes",
+					Value:     initializedVolumes,
+				}
+				patches = append(patches, patch1)
+			}
+		}
+
+		if len(patches) == 0 {
+			continue
+		}
+
+		jsonPatchJSON, err := json.Marshal(patches)
+		if err != nil {
+			return err
+		}
+
+		constantPatch := crClient.RawPatch(types.JSONPatchType, jsonPatchJSON)
+
+		// Since the pod status is updated from pod init container,
+		// set the field owner to "pod" for pod status updates.
+		setupLog.Info("Patching status with updated initialised volumes")
+
+		if err = client.Status().Patch(
+			context.TODO(), &aeroCluster, constantPatch, crClient.FieldOwner("pod"),
+		); err != nil {
+			return fmt.Errorf("error updating status: %v", err)
+		}
```

Comment:
- Update the comment
- Use context from above
- Shouldn't the field owner be `controller` in this case?

Reply:
[Here](https://github.com/aerospike/aerospike-kubernetes-operator/blob/f3bd4303156b43141f4861aab35d4d1e7b9a10f4/controllers/pod.go#L987) also we are updating patch from controller only with owner as pod and comment is regarding that only.